File: tournaments/constructSubmatrix/constructSubmatrix.py
def constructSubmatrix(matrix, rowsToDelete, columnsToDelete):
    
    res = []
    useRow = []
    useColumn = []

    for i in range(len(matrix)):
        useRow.append(True)
    for i in range(len(matrix[0])):
        useColumn.append(True)

    for i in range(len(rowsToDelete)):
        useRow[ rowsToDelete[i] ] = False
    for i in range(len(columnsToDelete)):
        useColumn[ columnsToDelete[i] ] = False

    for i in range(len(matrix)):
        if useRow[i]:
            res.append([])
            for j in range(len(matrix[0])):
                if useColumn[j]:
                    res[-1].append(matrix[i][j])

    return res

File: tournaments/constructSubmatrix/constructSubmatrix.js
function constructSubmatrix(matrix, rowsToDelete, columnsToDelete) {
    var res = [];
    var useRow = [];
    var useColumn = [];
  
    for (var i = 0; i < matrix.length; i++) {
      useRow.push(true);
    }
    for (var i = 0; i < matrix[0].length; i++) {
      useColumn.push(true);
    }
  
    for (var i = 0; i < rowsToDelete.length; i++) {
      useRow[rowsToDelete[i]] = false;
    }
    for (var i = 0; i < columnsToDelete.length; i++) {
      useColumn[columnsToDelete[i]] = false;
    }
  
    for (var i = 0; i < matrix.length; i++) {
      if (useRow[i]) {
        res.push([]);
        for (var j = 0; j < matrix[0].length; j++) {
          if (useColumn[j]) {
            res[res.length - 1].push(matrix[i][j]);
          }
        }
      }
    }
  
    return res;
  }
  